packages/x-components/src/components/result/base-result-image.vue

```diff
@@ -1,33 +1,36 @@
 <template>
   <picture ref="image" class="x-picture x-result-picture" data-test="result-picture">
-    <NoElement
+    <component
+      :is="animation"
       v-if="!hasImageLoaded && !hasAllImagesFailed"
       class="x-picture__image x-picture__image--placeholder"
     >
       <!-- eslint-disable-next-line max-len -->
       <!-- @slot (Required) Loading image content. It will be rendered while the real image is not loaded -->
       <slot name="placeholder" />
-    </NoElement>
-    <img
-      v-if="imageSrc"
-      v-show="hasImageLoaded"
-      @error="flagImageAsFailed"
-      @load="flagImageLoaded"
-      :alt="result.name"
-      :src="imageSrc"
-      class="x-picture__image x-result-picture__image"
-      data-test="result-picture__image"
-    />
-    <NoElement v-else-if="hasAllImagesFailed" class="x-picture__image x-picture__image--fallback">
-      <!--
-      Vue styleguidist doesn't generate slot docs for v-else and v-else-if conditions
-      due to a bug https://github.com/vuejs/vue/pull/10286.
-      TODO - Bump styleguidist version when the fix branch is merged and a new version released.
-      -->
-      <!-- eslint-disable-next-line max-len -->
-      <!-- @slot (Required) Fallback image content. It will be rendered when all the images failed -->
-      <slot name="fallback" />
-    </NoElement>
+    </component>
+    <component :is="animation">
+      <img
+        v-if="imageSrc"
+        v-show="hasImageLoaded"
+        @error="flagImageAsFailed"
+        @load="flagImageLoaded"
+        :alt="result.name"
+        :src="imageSrc"
+        class="x-picture__image x-result-picture__image"
+        data-test="result-picture__image"
+      />
+      <NoElement v-else-if="hasAllImagesFailed" class="x-picture__image x-picture__image--fallback">
```

Comment:
Why is this a `NoComponent`? can't be a `<template>`?

Reply:
Are you asking me 🤣? 
![image](https://user-images.githubusercontent.com/68222542/134009103-64fdb4a2-dfac-4d9e-aee7-9a8bea99fbed.png)

I remember discussing about using `NoElement` to add classes to slots, and I guess that's the case here, adding the design system classes to the slot content. But I'm not sure if there are other reasons.

Reply:
Don't play with my poor memory!

I guess it was for the classes as you say.